components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -142,30 +129,91 @@ - (UIColor *)titleColorForState:(UIControlState)state {
 
 #pragma mark - UIView
 
-// TODO(#7645): Temporary layout until
-// https://github.com/material-components/material-components-ios/issues/7645 lands
 - (void)layoutSubviews {
   [super layoutSubviews];
 
-  if (self.items.count == 0) {
-    return;
+  CGFloat availableWidth = self.bounds.size.width;
+  CGFloat maxWidth = 0;
+  for (UIView *itemView in self.containerView.arrangedSubviews) {
+    CGSize contentSize = itemView.intrinsicContentSize;
+    if (contentSize.width > maxWidth) {
+      maxWidth = contentSize.width;
+    }
   }
-  CGSize boundsSize = CGSizeMake(CGRectGetWidth(self.bounds), CGRectGetHeight(self.bounds));
-  CGSize itemSize = CGSizeMake(boundsSize.width / self.items.count, boundsSize.height);
+  CGFloat requiredWidth = maxWidth * self.items.count;
+  BOOL canBeJustified = availableWidth > requiredWidth;
+  self.containerView.distribution = canBeJustified ? UIStackViewDistributionFillEqually
+                                                   : UIStackViewDistributionFillProportionally;
+}
 
-  CGPoint itemOrigin = CGPointMake(CGRectGetMinX(self.bounds), CGRectGetMinY(self.bounds));
-  for (UIView *itemView in self.itemViews) {
-    itemView.frame = CGRectMake(itemOrigin.x, itemOrigin.y, itemSize.width, itemSize.height);
-    itemOrigin = CGPointMake(itemOrigin.x + itemSize.width, itemOrigin.y);
+- (void)updateConstraints {
+  [super updateConstraints];
+
+  if (self.viewConstraints) {
+    return;
   }
+
+  NSMutableArray *constraints = [NSMutableArray array];
+  [constraints addObjectsFromArray:@[
+    [_containerView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
+    [_containerView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor],
+    [_containerView.widthAnchor constraintGreaterThanOrEqualToAnchor:self.widthAnchor],
+    [_containerView.topAnchor constraintEqualToAnchor:self.topAnchor],
+    [_containerView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor],
+  ]];
+
+  [self addConstraints:constraints];
+  self.viewConstraints = constraints;
 }
 
 - (CGSize)intrinsicContentSize {
-  return CGSizeMake(UIViewNoIntrinsicMetric, kMinHeight);
+  CGFloat totalWidth = 0;
+  CGFloat maxHeight = 0;
+  for (UIView *itemView in self.containerView.arrangedSubviews) {
+    CGSize contentSize = itemView.intrinsicContentSize;
+    totalWidth += contentSize.width;
+    if (contentSize.height > maxHeight) {
+      maxHeight = contentSize.height;
+    }
+  }
+  return CGSizeMake(totalWidth, MAX(kMinHeight, maxHeight));
 }
 
 - (CGSize)sizeThatFits:(CGSize)size {
-  return CGSizeMake(size.width, MAX(size.height, kMinHeight));
+  CGSize intrinsicSize = self.intrinsicContentSize;
+  return CGSizeMake(MIN(intrinsicSize.width, size.width), MAX(intrinsicSize.height, size.height));
```

Comment:
I'm curious about this logic. If the current bounds is `CGRectZero` and `sizeToFit` is called, then this view would only resize its height. Is that expected?  Should it not attempt to provide a fit size for its contents as a minimum, but allow a larger width?

Reply:
I am thinking about since it is a scrollable view thus this should be fits in any width.

Reply:
CGZero is a extremely case, but if a small bounds is provided, the scrollable view should also fit itself in

Reply:
That's true, but the fit size would (ideally) be the size of its content. I think the best thing to conform to the documented behavior of `sizeThatFits:` is to return the "fit size".  In this case, it's probably the the size of the justified layout (rather than the intrinsic content size).  Does that make sense?

Whether it's kept like this or changed, I think it's worth adding a unit test that calls `sizeToFit:` with a bounds of `CGRectZero` and make sure it gets back a non-zero width and non-zero height.  Something similar for a starting bounds bigger than necessary?

Reply:
Sure. Fixed applied. I also refactored the intrisicSize to justified width instead of scrollable width.